src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray.cs

```diff
@@ -145,16 +145,19 @@ public static ImmutableArray<T> CreateRange<T>(IEnumerable<T> items)
         [Pure]
         public static ImmutableArray<T> Create<T>(params T[]? items)
         {
-            if (items == null)
+            int length = items?.Length ?? 0;
+            if (length == 0)
```

Comment:
```suggestion
            if (items == null || items.Length == 0)
```

and then you can omit `!` below. We prefer to write the code without nullable `!` if it is not too hard.

I do not think you need to worry about fetching `items.Length` just once. It is trivial for the JIT to optimize out the redundant fetch.

Reply:
Fixed, thanks for the 👀